rp2040-hal/src/spi.rs

```diff
@@ -53,6 +54,16 @@ impl From<&embedded_hal::spi::Mode> for Mode {
     }
 }
 
+/// SPI frame format
+pub enum FrameFormat {
```

Comment:
Can we make it derive a few useful things like
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[cfg_attr(feature = "defmt", derive(defmt::Format))]
pub enum FrameFormat {
```

Reply:
I added most of these, but if you want Debug and defmt::Format, I'll have to bring the newtype back.